rinja_derive/src/tests.rs

```diff
@@ -73,11 +75,13 @@ impl ::std::fmt::Display for Foo {{
     compare(
         "{% if let Some(s) = s %}{{ s }}{% endif %}",
         r#"if let Some(s,) = &self.s {
-    ::std::write!(
-        writer,
-        "{expr0}",
-        expr0 = &(&&::rinja::filters::AutoEscaper::new(&(s), ::rinja::filters::Text)).rinja_auto_escape()?,
-    )?;
+    match (
+        &((&&::rinja::filters::AutoEscaper::new(&(s), ::rinja::filters::Text)).rinja_auto_escape()?),
+    ) {
+        (expr0,) => {
+            (&&::rinja::filters::Writable(expr0)).rinja_write(writer)?;
+        }
+    }
```

Comment:
Using a `match` instead of `{…}` block, we do not need to fear accidental shadowing, and the lifetimes are more easy to understand, IMO.

Reply:
Why not `if let` in this case?

Reply:
Rust's `derive(Debug, Clone)` etc. generate code using `match`es, so I did not really think about it. I guess it's OK either way?

Reply:
`if let` is shorter but it's a nit/style prefer, I won't block this PR on this (because I don't really care :laughing:).